src/sentences/utils.rs

```diff
@@ -49,6 +49,43 @@ pub(crate) fn parse_hms(i: &str) -> IResult<&str, NaiveTime> {
     )(i)
 }
 
+/// The number of milliseconds in a second.
+const MILLISECS_PER_SECOND: i64 = 1000;
+/// The number of milliseconds in a minute.
+const MILLISECS_PER_MINUTE: i64 = 60000;
+/// The number of milliseconds in a hour.
+const MILLISECS_PER_HOUR: i64 = 3600000;
+
+pub(crate) fn parse_duration_hms(i: &str) -> IResult<&str, Duration> {
+    map_res(
+        tuple((
+            map_res(take(2usize), parse_num::<i64>),
```

Comment:
- [ ] Please use a different value for the number, e.g. `u8` will also be sufficient.

Reply:
i64 makes easier following conversion to Duration (as Duration accepts i64).

Reply:
It would be better to fail value parsing as early as possible.

And Yes, Duration takes i64, however, we could easily convert it with From: https://doc.rust-lang.org/std/primitive.i64.html#impl-From%3Cu8%3E-for-i64

It's better to parse a mostly correct value, instead of very wrong value. parsing i64 could also take longer and can have bigger impact on constraint device since this first parser can actually succeed event with e.g. negative value.

Even if the architecture is 32 bit, 64 bit numbers are still supported but not by the hardware itself, so are less performant.